### DISCUSSION TITLE
Does Flask SocketIO (on eventlet) modify regular flask routes?

### DISCUSSION THREAD

**Discussion by theahura** [miguelgrinberg/Flask-SocketIO#1584] [Q&A]
**Your question**
Hey Miguel, 

Thank you for being so responsive! Complete game changer when it comes to using this library :smile: 

I had another question. 
Some background: I'm running Flask SocketIO using eventlet as a backend, on kubernetes. I'm noticing that my kubernetes liveness probes end up failing whenever I use the socket. I would have thought that any 'normal' flask routes would be unaffected by the socket. 

1) What's going on under the hood? I know eventlet makes everything into an eventloop -- does that impact normal flask routes as well?
2) Is there a way to have normal flask routes behave as normal, and have sockets behave separately/in isolation? 

Thanks in advance!

**Logs**
NA

**Comment by miguelgrinberg** [comment]
> I know eventlet makes everything into an eventloop -- does that impact normal flask routes as well?

Yes, the whole process runs in async mode.

> Is there a way to have normal flask routes behave as normal, and have sockets behave separately/in isolation?

You can separate your application into two servers that run separately.
